### PR TITLE
Calls: Switch camera lights off when muting

### PR DESCRIPTION
**Pull request by MidhunSureshR**

### CODE REVIEW

File: src/matrix/calls/group/GroupCall.ts [hunk 1]
```diff
@@ -125,6 +127,7 @@ export class GroupCall extends EventEmitter<{change: never}> {
     }
 
     get localMedia(): LocalMedia | undefined { return this.joinedData?.localMedia; }
+    get localPreviewMedia(): LocalMedia | undefined { return this.joinedData?.localPreviewMedia; }
```
Comment (bwindels) [inline src/matrix/calls/group/GroupCall.ts:130]: I think we can remove the `localMedia` getter. It should only be used from previewing in the UI.

File: src/matrix/calls/group/GroupCall.ts [hunk 2]
```diff
@@ -165,10 +168,12 @@ export class GroupCall extends EventEmitter<{change: never}> {
         const membersLogItem = logItem.child("member connections");
         const localMuteSettings = new MuteSettings();
         localMuteSettings.updateTrackInfo(localMedia.userMedia);
+        const localPreviewMedia = localMedia.asPreview();
```
Comment (bwindels) [inline src/matrix/calls/group/GroupCall.ts:171]: I think we might need to do this as well in `setMedia`?

**Review by bwindels — COMMENTED** [review]
Thanks, looks like a better approach indeed. Just a few comments.

File: src/matrix/calls/group/GroupCall.ts (hunk 1 above)
Comment (MidhunSureshR) [reply src/matrix/calls/group/GroupCall.ts:130]: It's used in CallViewModel.toggleCamera() and toggleMicrophone() to check if tracks are available

File: src/matrix/calls/group/GroupCall.ts [hunk 3]
```diff
@@ -200,6 +200,8 @@ export class GroupCall extends EventEmitter<{change: never}> {
         if ((this._state === GroupCallState.Joining || this._state === GroupCallState.Joined) && this.joinedData) {
             const oldMedia = this.joinedData.localMedia;
             this.joinedData.localMedia = localMedia;
+            this.localPreviewMedia?.dispose();
```
Comment (bwindels) [inline src/matrix/calls/group/GroupCall.ts]: Slightly confusing to use the public getter here for accessing and the private member for assigning, but I guess that works.

**Pull request merged**